src/cli_config.rs

```diff
@@ -113,84 +126,180 @@ pub fn cli_parse() -> CliConfig {
         .author(crate_authors!("\n"))
         .about(crate_description!())
         .setting(AppSettings::ColoredHelp)
-        .arg(Arg::with_name("udp-address")
-            .short("u")
-            .long("udp-address")
-            .help("UDP address to run DHT node")
-            .takes_value(true)
-            .required_unless("tcp-address"))
-        .arg(Arg::with_name("tcp-address")
-            .short("t")
-            .long("tcp-address")
-            .help("TCP address to run TCP relay")
-            .multiple(true)
-            .takes_value(true)
-            .use_delimiter(true)
-            .required_unless("udp-address"))
-        .arg(Arg::with_name("secret-key")
-            .short("s")
-            .long("secret-key")
-            .help("DHT secret key. Note that you should not pass the key via \
+        .subcommand(SubCommand::with_name("config")
+            .arg(Arg::with_name("cfg-file")
+                .short("f")
+                .long("cfg-file")
+                .help("Load settings from saved config file. \
+                 Config file format is YAML")
+                .takes_value(true))
+            .arg(Arg::with_name("secret-key")
+                .short("s")
+                .long("secret-key")
+                .help("DHT secret key. Note that you should not pass the key via \
+                   arguments due to security reasons. Use this argument for \
+                   test purposes only. In the real world use the environment \
+                   variable instead")
+                .takes_value(true)
+                .env("TOX_SECRET_KEY")
+                .hidden(true)))
+        .subcommand(SubCommand::with_name("args")
```

Comment:
Well, do we really need to: `tox-node args --args1 --arg2`? looks redundant

Reply:
As I know, if we use one subcommand, the other args also must be used as other subcommand.

Reply:
Did you try?

Reply:
Yes I tried it.

Reply:
Not necessary: https://docs.rs/clap/2.32.0/clap/enum.AppSettings.html#variant.SubcommandsNegateReqs

Reply:
done